src/lib/features/project/project-service.ts

```diff
@@ -310,7 +320,25 @@ export default class ProjectService {
             }),
         );
 
-        await enableChangeRequestsForSpecifiedEnvironments();
+        if (
+            this.isEnterprise &&
+            this.flagResolver.isEnabled('createProjectWithEnvironmentConfig')
+        ) {
+            if (data.changeRequestEnvironments) {
+                await this.validateEnvironmentsExist(
+                    data.changeRequestEnvironments,
+                );
+                const changeRequestEnvironments =
+                    await enableChangeRequestsForSpecifiedEnvironments(
+                        data.changeRequestEnvironments,
+                    );
+
+                data.changeRequestEnvironments =
+                    changeRequestEnvironments ?? [];
+            } else {
+                data.changeRequestEnvironments = [];
+            }
+        }
```

Comment:
This is the meat of the change. I've done a few things here:

1. Only call this method if we're on enterprise (and if the flag is enabled)
2. Validate the data in OSS before passing it to the function. That keeps all validation in one place and we can safely assume that all envs passed into the function are valid.
3. Assume that the function will return the list of environments and their required approvals. 
4. If we're on enterprise and you didn't provide any change request environments, then we'll return an empty list to indicate that no envs were set up.

This means that when we call this in enterprise, there's less worrying about logic and that the data is valid. Instead, it's a function that takes an input and yields an output. In addition to putting all the heavy logic in one place, it also makes it easier to test all of this in one spot.